### PR TITLE
chore: move table config to a slice

### PR DESCRIPTION
**Pull request by magnew**
### Description:

- Move table config to a slice
- Make table config options a component
- Move barchart config component out of viz directory 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/sqlRunner/store/tableVisSlice.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            if (config && config.columns[reference]) {
+                config.columns[reference].label = label;
+            }
+        },
+    },
+    extraReducers: (builder) => {
+        builder.addCase(setInitialResultsAndSeries, (state, action) => {
+            if (state.config === undefined) {
+                // TODO: this should come from the transformer
+                const columns = Object.keys(action.payload[0]).reduce<
```
Comment (magnew) [inline packages/frontend/src/features/sqlRunner/store/tableVisSlice.ts:28]: This is duplicated per table and should go on the transformer, but we are still working out the design for transformers. We can move this when that's ready.

File: packages/frontend/src/features/sqlRunner/components/RightSidebar.tsx [hunk 2]
```diff
@@ -11,24 +11,20 @@ import {
 import { IconLayoutSidebarRightCollapse } from '@tabler/icons-react';
 import { type Dispatch, type FC, type SetStateAction } from 'react';
 import MantineIcon from '../../../components/common/MantineIcon';
-import { EditableText } from '../../../components/VisualizationConfigs/common/EditableText';
+
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/components/RightSidebar.tsx:14]: can you remove this empty line?

File: packages/frontend/src/features/sqlRunner/store/index.ts [hunk 3]
```diff
@@ -3,12 +3,15 @@
 import type * as rtk from '@reduxjs/toolkit';
 import { configureStore } from '@reduxjs/toolkit';
 import { barChartConfigSlice } from './barChartSlice';
+import { tableVisSlice } from './tableVisSlice';
+
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/store/index.ts:7]: here as well

File: packages/frontend/src/features/sqlRunner/store/sqlRunnerSlice.ts [hunk 4]
```diff
@@ -104,24 +100,10 @@ export const sqlRunnerSlice = createSlice({
                 {},
             );
             // Set static results table
+            // TODO: should this be in a separate slice?
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/store/sqlRunnerSlice.ts:103]: i see the benefit 👍

**Review by joaoviana — APPROVED** [review]

**Pull request merged**